docs/victory-candlestick/ecology.md

```diff
@@ -0,0 +1,274 @@
+
+
+The `eventKey` may optionally be used to select a single element by index rather than an entire set. The `eventHandlers` object should be given as an object whose keys are standard event names (i.e. `onClick`) and whose values are event callbacks. The return value of an event handler is used to modify elements. The return value should be given as an object or an array of objects with optional `eventKey` and `target` keys, and a `mutation` key whose value is a function. The `eventKey` and `target` keys will default to values corresponding to the element the event handler was attached to. The `mutation` function will be called with the calculated props for the individual selected element (_i.e._ a single candlestick), and the object returned from the mutation function will override the props of the selected element via object assignment. VictoryCandlestick may also be used with the `VictorySharedEvents` wrapper.
+
+```playground
+<VictoryCandlestick
+  height={500}
+  padding={75}
+  data={[
+    {x: new Date(2016, 6, 1), open: 5, close: 10, high: 15, low: 0},
+    {x: new Date(2016, 6, 2), open: 15, close: 10, high: 20, low: 5, label: "hello"},
+    {x: new Date(2016, 6, 3), open: 15, close: 20, high: 25, low: 10},
+    {x: new Date(2016, 6, 4), open: 20, close: 25, high: 30, low: 15, label: "there"},
+    {x: new Date(2016, 6, 5), open: 30, close: 25, high: 35, low: 20}
+  ]}
+  events={[{
+    target: "labels",
+    eventHandlers: {
+      onClick: () => {
+        return [
+          {
+            mutation: (props) => {
+              return {
+                style: merge({}, props.style.labels, {fill: "orange"})
+              };
+            }
+          }
+        ];
+      }
+    }
+  },
+  {
+    target: "data",
+    eventHandlers: {
+      onClick: () => {
+        return [
+          {
+            mutation: (props) => {
+              return {
+                style: merge({}, props.style, {fill: "blue"})
+              };
+            }
+          }
+        ];
+      }
+    }
+  }]}
+/>
+```
+
+### Animating
+
+VictoryCandlestick animates with [VictoryAnimation][] as data changes when an `animate` prop is provided.
+VictoryCandlestick defines a set of default transition behaviors for entering and exiting data nodes.
+Provide `onExit` and `onEnter` via the animate prop to define custom enter and exit transitions.
+Values returned from `before` and `after` functions will alter the data prop of entering or exiting nodes.
+
+```playground_norender
+class App extends React.Component {
+   constructor(props) {
+    super(props);
+    this.state = {
+      data: this.getData(),
+    };
+  }
+
+  getData() {
+    const dataSets = [
+      [
+        {x: 1, open: 5, close: 10, high: 15, low: 0},
+        {x: 2, open: 15, close: 10, high: 20, low: 5},
```

Comment:
Can you define these in a more concise format. It makes sense to keep the x values evenly spaced like that, but maybe something like...
```
getData() {
    const n = random(4, 10)
    return range(n).map((i) => {
      return {
        x: i,
        open: random(10, 20),
        close: random(10, 20),
        high: random(20, 30),
        low: random(0, 10)
      };
    });
  }
```